### PR TITLE
Send sticker

### PR DESCRIPTION
**Pull request by wolkemann**
Implemented the backend and frontend logic for the sticker's sending.

- The users tab in the database now has a new line called `stickers`, which is an array that will store all the sticker collected.
- the item in the array are inserted using the spread syntax `stickers: [...OldArray, newElement]`, and in the backend the PATCH method is used to updated the data.

I don't know if this is the right why to update an array inside MongoDB, but at the moment it works

### CODE REVIEW

File: pages/api/users/[userId].js [hunk 1]
```diff
@@ -7,15 +7,25 @@ export default async function handler(request, response) {
...
-        const getUser = await User.findById(userId).exec();
         response.status(200).json(getUser);
         break;
 
+      case "PATCH":
+        const updatedUser = await User.findByIdAndUpdate(
+          userId,
+          {
+            ...request.body,
+          },
```
Comment (djfarly) [inline pages/api/users/[userId].js:23]: I think you can get away without copying the body here :)

```suggestion
          request.body,
```

File: pages/inbox/browse.js [hunk 2]
```diff
@@ -14,15 +14,39 @@ export const UserContext = createContext();
...
+
+    const updateEmojiCollection = await fetch(`/api/users/${reply.authorId}`, {
+      method: "PATCH",
+      headers: { "content-type": "application/json" },
+      body: JSON.stringify({
+        stickers: [...Reciever.stickers, selectedSticker],
+      }),
+    });
+
+    const updatedOk = await updateEmojiCollection.json();
```
Comment (djfarly) [inline pages/inbox/browse.js:42]: I don't think you're doing anything with `updatedOk` here…

File: pages/inbox/browse.js [hunk 3]
```diff
@@ -14,15 +14,39 @@ export const UserContext = createContext();
...
+    });
+    const Reciever = await fetchReciever.json();
+
+    const updateEmojiCollection = await fetch(`/api/users/${reply.authorId}`, {
+      method: "PATCH",
+      headers: { "content-type": "application/json" },
+      body: JSON.stringify({
+        stickers: [...Reciever.stickers, selectedSticker],
+      }),
+    });
```
Comment (djfarly) [inline pages/inbox/browse.js:40]: I think you can do this in one operation:

see: https://www.mongodb.com/docs/manual/reference/operator/update/push/

```suggestion
    const updatedUserResponse = await fetch(`/api/users/${reply.authorId}`, {
      method: "PATCH",
      headers: { "content-type": "application/json" },
      body: JSON.stringify({
        '$push': { 'stickers': selectedSticker },
      }),
    });
    
    // if you need the user afterwards:
    // const updatedUser = updatedUserResponse.json();
```

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Fede,

Really great work!

I don't know if the data structure you chose is sufficient because you are simply storing stickers the user recieved in a list without knowing where they came from. If that is what you want you're good to go.

I added one suggestion on how to simplify your code.

Keep up the great work!

**Comment by wolkemann** [comment]
> I don't know if the data structure you chose is sufficient because you are simply storing stickers the user recieved in a list without knowing where they came from. If that is what you want you're good to go.

Yes the data structure was a little bit too simple so I updated it. Now inside the array there would be objetcs with this structure: `{url: "stickers.svg", sender: "senderNickname"}`.

Thank you again, Coach :)

**Pull request merged**